src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -195,29 +187,32 @@
   [aliases context config]
   {:pre  [(or (coll? aliases) (string? aliases) (nil? aliases))]
    :post [(= (type %) (type aliases))]}
-  (if (coll? aliases)
-    (vec (map #(anonymize-leaf % :title context config) aliases))
-    (if (string? aliases)
-      (anonymize-leaf aliases :title context config)
-      aliases)))
+  (when-not (nil? aliases)
+    (if (coll? aliases)
+      (mapv #(anonymize-leaf % :title context config) aliases)
+      (anonymize-leaf aliases :title context config))))
 
 (defn anonymize-parameter
   "Anonymize a parameter/value pair"
   [parameter context config]
   {:pre  [(coll? parameter)]
    :post [(coll? %)]}
-  (let [[key val]  parameter
-        newcontext (assoc-in context ["parameter-value"] val)]
+  (let [[key val] parameter
+        newcontext (assoc context "parameter-value" val)]
     (case key
       ;; Metaparameters are special
-      ("stage" "tag")        [key (anonymize-leaf val :title newcontext config)]
-      "alias"                [key (anonymize-aliases val newcontext config)]
+      ("stage" "tag")
+      [key (anonymize-leaf val :title newcontext config)]
+      "alias"
+      [key (anonymize-aliases val newcontext config)]
+
       ;; References get randomized in a special way
-      ("require" "before"
-       "notify" "subscribe") [key (anonymize-references val newcontext config)]
-       ;; Everything else gets anonymized as per normal
-       [(anonymize-leaf key :parameter-name newcontext config)
-        (anonymize-leaf val :parameter-value newcontext config)])))
+      ("require" "before" "notify" "subscribe")
```

Comment:
TIL you can pass a list to `case`. Really not in love with the syntax, but them's the breaks.